src/main/java/org/jabref/logic/integrity/ValidBibtexKeyChecker.java

```diff
@@ -17,7 +18,7 @@
             return Optional.of(Localization.lang("empty BibTeX key"));
         }
 
-        String cleaned = BibtexKeyGenerator.cleanKey(value);
+        String cleaned = BibtexKeyGenerator.cleanKey(value, Globals.prefs.getUnwantedCharacters());
```

Comment:
Here 'ValidBibtexKeyChecker' cannot depend on 'org.jabref.Globals', but if i pass it using be 'BibtexKeyPatternPreferences.java', 'FieldCheckers' class have to take 'BibtexKeyPatternPreferences' instance also. would this solution be correct?